src/pool/mempool.rs

```diff
@@ -58,35 +58,34 @@ impl AccountManager {
         Self { accounts }
     }
 
-    pub fn start<SP>(&self, rt_handle: &Handle, eth_provider: &'static EthDataProvider<SP>)
+    pub fn start<SP>(&self, rt_handle: &Handle, eth_client: &'static EthClient<SP>)
     where
-        SP: starknet::providers::Provider + Send + Sync + 'static,
+        SP: starknet::providers::Provider + Send + Sync + Clone + 'static,
```

Comment:
does it need to be clone? I don't see this used

Reply:
Here too, it's due to the other constraint...